### PR TITLE
fix: missing account verification modal

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- Refactored the `useAuthRecovery` to manage verification as well (and renamed the file).
- Managed the verification messages including showing the modal if it is finished.

Preview:
image

There are two ways for you to see this, whether you are already logged in or not. Once the former occurred, it displays the below modal (without the left side and changing the header). Though I feel like it is a bit empty even after I made the icon bigger.
Also the text we will have to wait for the other team's response (I will send one for it in auth channel).
image

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in a companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-597 #done

### CODE REVIEW

File: packages/shared/src/hooks/useAuthVerificationRecovery.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      onSuccess: checkErrorMessage,
+    },
+  );
+
+  useQuery(
+    [{ type: 'verification', flow: router?.query.flow as string }],
+    ({ queryKey: [{ flow }] }) => getKratosFlow(AuthFlow.Verification, flow),
+    {
+      ...disabledRefetch,
+      enabled: !!router?.query.flow && !router?.query.recovery,
```
Comment (rebelchris) [inline packages/shared/src/hooks/useAuthVerificationRecovery.ts:68]: Is recovery correct here?
Both queries fire for the same thing?

File: packages/shared/src/components/auth/EmailVerified.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+function EmailVerified({
+  hasUser,
+  children,
+  onClose,
+}: EmailVerifiedProps): ReactElement {
+  return (
+    <>
+      <AuthModalHeader
+        title={hasUser ? 'Email address verified' : 'Log in to daily.dev'}
```
Comment (rebelchris) [inline packages/shared/src/components/auth/EmailVerified.tsx:21]: Not quite sure why this title is dynamic as the text always implies you are logged in?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Nice, only some small remarks.

File: packages/shared/src/components/auth/EmailVerified.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/auth/EmailVerified.tsx:21]: Yes. There are two states where you can click the verification link. Whether you are logged in or not is valid.

File: packages/shared/src/hooks/useAuthVerificationRecovery.ts (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/hooks/useAuthVerificationRecovery.ts:68]: The other one is recovery, this one is for verification.

**Review by rebelchris — APPROVED** [review]

**Pull request merged**